dotCMS/src/main/java/com/dotmarketing/business/PermissionBitAPIImpl.java

```diff
@@ -325,12 +320,16 @@ public boolean doesUserHavePermission(Permissionable permissionable, int permiss
 			permissionType=PERMISSION_EDIT;
 		}
 		
-		// http://jira.dotmarketing.net/browse/DOTCMS-6943
-		// everybody should be able to use file structures
-        if (permissionable instanceof Structure
-                && (permissionType==PERMISSION_WRITE || permissionType==PERMISSION_PUBLISH)
-                && ((Structure)permissionable).getStructureType()==Structure.STRUCTURE_TYPE_FILEASSET)
-            return true;
+        // everybody should be able to use file content types
+        if (permissionable instanceof Structure || permissionable instanceof ContentType) {
+            final Structure contentType = (permissionable instanceof ContentType)
+                            ? new StructureTransformer(ContentType.class.cast(permissionable)).asStructure()
+                            : Structure.class.cast(permissionable);
+            if ((PERMISSION_WRITE == permissionType || PERMISSION_PUBLISH == permissionType)

```

Comment:
you can simplify that with just a return 
```
return ((PERMISSION_WRITE == permissionType || PERMISSION_PUBLISH == permissionType) 
&& BaseContentType.FILEASSET.getType() == contentType.getStructureType());

Reply:
The method returns "true" if that condition applies. Otherwise, it continues, it doesn't return false so the existing code cannot be replaced with the suggested code.